service/http/tls.go

```diff
@@ -25,14 +27,75 @@ type TLSArguments struct {
 	MinVersion       TLSVersion        `river:"min_version,attr,optional"`
 	MaxVersion       TLSVersion        `river:"max_version,attr,optional"`
 
-	// TODO(rfratto): windows certificate filter.
+	// Windows Certificate Filter
+	WindowsFilter *WindowsCertificateFilter `river:"windows_certificate_filter,block,optional"`
+}
+
+// WindowsCertificateFilter represents the configuration for accessing the Windows store
+type WindowsCertificateFilter struct {
+	Server *WindowsServerFilter `river:"server,block"`
+	Client *WindowsClientFilter `river:"client,block"`
+}
+
+// WindowsClientFilter is used to select a client root CA certificate
+type WindowsClientFilter struct {
+	IssuerCommonNames []string `river:"issuer_common_names,attr,optional"`
+	SubjectRegEx      string   `river:"subject_regex,attr,optional"`
+	TemplateID        string   `river:"template_id,attr,optional"`
+}
+
+// WindowsServerFilter is used to select a server certificate
+type WindowsServerFilter struct {
+	Store             string        `river:"store,attr,optional"`
+	SystemStore       string        `river:"system_store,attr,optional"`
+	IssuerCommonNames []string      `river:"issuer_common_names,attr,optional"`
+	TemplateID        string        `river:"template_id,attr,optional"`
+	RefreshInterval   time.Duration `river:"refresh_interval,attr,optional"`
+}
+
+var _ river.Defaulter = (*WindowsServerFilter)(nil)
+
+// SetToDefault sets the default for WindowsServerFilter
+func (wcf *WindowsServerFilter) SetToDefault() {
+	wcf.RefreshInterval = 5 * time.Minute
 }
 
 var _ river.Validator = (*TLSArguments)(nil)
 
-// Validate returns whether args is valid. It checks that mutually exclusive
-// fields are not both set, and that required fields are set.
+// Validate returns whether args is valid.
 func (args *TLSArguments) Validate() error {
+	if args.WindowsFilter == nil {
+		return args.validateTLS()
+	}
+	return args.validateWindowsCertificateFilterTLS()
+}
+
+// validateWindowsCertificateFilterTLS validates the Windows Certificate filter details.
+func (args *TLSArguments) validateWindowsCertificateFilterTLS() error {
+	switch {
+	case len(args.Cert) > 0:
+	case len(args.Key) > 0:
+	case len(args.CertFile) > 0:
+	case len(args.ClientCA) > 0:
+	case len(args.ClientCAFile) > 0:
+	case len(args.KeyFile) > 0:
+		return fmt.Errorf("cannot specify any key, certificate or CA when using windows certificate filter")
+	}
```

Comment:
there is no fallthrough by default in Go, so unless Im mistaken the error is returned only when len(args.KeyFile) is > 0 but in the other cases it's not returning the error?
Maybe you meant this instead:
```
switch {
case len(args.Cert) > 0,
     len(args.Key) > 0,
     len(args.CertFile) > 0,
     len(args.ClientCA) > 0,
     len(args.ClientCAFile) > 0,
     len(args.KeyFile) > 0:
	return fmt.Errorf("cannot specify any key, certificate or CA when using windows certificate filter")
}
```